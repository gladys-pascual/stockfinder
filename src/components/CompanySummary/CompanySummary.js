import "./CompanySummary.scss";
import PropTypes from "prop-types";

const CompanySummary = ({
  companyName,
  ticker,
  industry,
  currentPrice,
  openPrice,
  prevClose,
  high,
  low,
  marketCap,
  ipo,
}) => {
  //Determine if current price is below or above the opening price
  const profitOrLoss = currentPrice - openPrice;

  const currentPricePnL = (profitOrLoss) => {
    if (Math.sign(profitOrLoss) === 1) {
      return "profit";
    } else {
      return "loss";
    }
  };

  //Convert Market Cap prices to million, billion or trillion
  const convertMarketCap = (price) => {
    if (price.toFixed(0).toString().length <= 3) {
      return `${price.toFixed(1)} M`;
    } else if (
      price.toFixed(0).toString().length > 3 &&
      price.toFixed(0).toString().length <= 6
    ) {
      return `${(price / 1000).toFixed(1)} B`;
    } else if (price.toFixed(0).toString().length > 6) {
      return `${(price / 1000000).toFixed(1)} T`;
    }
  };

  // Convert IPO date to month and year
  const convertIpo = (date) => {
    const stringDate = date.split("");
    const year = stringDate.slice(0, 4).join("");
    let month = stringDate.slice(5, 7).join("");
    if (month === "01") {
      month = "Jan";
    } else if (month === "02") {
      month = "Feb";
    } else if (month === "03") {
      month = "Mar";
    } else if (month === "04") {
      month = "Apr";
    } else if (month === "05") {
      month = "May";
    } else if (month === "06") {
      month = "Jun";
    } else if (month === "07") {
      month = "Jul";
    } else if (month === "08") {
      month = "Aug";
    } else if (month === "09") {
      month = "Sep";
    } else if (month === "10") {
      month = "Oct";
    } else if (month === "11") {
      month = "Nov";
    } else if (month === "12") {
      month = "Dec";
    }
    return `${month} ${year}`;
  };

  return (
    <section className="company-summary">
      <div className="summary-wrapper">
        <div className="lhs">
          <div className="company-header">
            <h1 className="ticker">{ticker}</h1>
            <div>
              <p className="company-name">{companyName}</p>
              <p className={industry === "N/A" ? "no-display" : "industry"}>
                {industry}
              </p>
            </div>
          </div>

          <p className={`current-price-label ${currentPricePnL(profitOrLoss)}`}>
            Current Price
          </p>
          <div className="price-and-pnl">
            <h1 className={`current-price ${currentPricePnL(profitOrLoss)}`}>
              {currentPrice}
            </h1>
            <p className={`${currentPricePnL(profitOrLoss)} pnl`}>
              {Math.sign(profitOrLoss) === 1
                ? `+${((profitOrLoss / currentPrice) * 100).toFixed(1)} %`
                : `${((profitOrLoss / currentPrice) * 100).toFixed(1)} %`}
            </p>
          </div>
        </div>
        <div className="rhs">
          <div className="prices">
            <div className="prices-lhs">
              <p className="prices-heading">OPEN</p>
              <p className="prices-text">{openPrice}</p>
            </div>
            <div className="prices-rhs">
              <p className="prices-heading">PREV CLOSE</p>
              <p className="prices-text">{prevClose}</p>
            </div>
          </div>
          <div className="prices">
            <div className="prices-lhs">
              <p className="prices-heading">HIGH</p>
              <p className="prices-text">{high}</p>
            </div>
            <div className="prices-rhs">
              <p className="prices-heading">LOW</p>
              <p className="prices-text">{low}</p>
            </div>
          </div>
          <div className="prices">
            <div className="prices-lhs">
              <p className="prices-heading">MARKET CAP</p>
              <p className="prices-text">{convertMarketCap(marketCap)}</p>
            </div>
            <div className="prices-rhs">
              <p className="prices-heading">IPO</p>
              <p className="prices-text">{convertIpo(ipo)}</p>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

CompanySummary.propTypes = {
  companyName: PropTypes.string.isRequired,
  ticker: PropTypes.string.isRequired,
  industry: PropTypes.string.isRequired,
  currentPrice: PropTypes.number.isRequired,
  openPrice: PropTypes.number.isRequired,
  prevClose: PropTypes.number.isRequired,
  high: PropTypes.number.isRequired,
  low: PropTypes.number.isRequired,
  marketCap: PropTypes.number.isRequired,
  ipo: PropTypes.string.isRequired,
};

export default CompanySummary;
