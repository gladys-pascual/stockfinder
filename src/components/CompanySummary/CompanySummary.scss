@import "../../assets/variables.scss";

.company-summary {
  background-image: linear-gradient($theme-color, ($theme-color + 70%));
  display: flex;
  .summary-wrapper {
    padding: 12px 10%;
    display: flex;
    justify-content: space-between;
    .lhs,
    .rhs {
      width: 50vw;
    }
    .lhs {
      .company-header {
        display: flex;
        align-items: center;
        .ticker {
          @include headings;
          font-size: 3em;
          text-align: left;
          margin: 0px;
          padding-right: 12px;
        }
        .company-name,
        .industry {
          margin: 0 0 4px 0;
          font-size: 1.1em;
        }
        .company-name {
          letter-spacing: 2px;
          font-weight: 600;
        }
        .industry {
          font-style: italic;
          letter-spacing: 1px;
        }
      }
      .current-price {
        @include headings;
        text-align: left;
      }
    }
  }
}
