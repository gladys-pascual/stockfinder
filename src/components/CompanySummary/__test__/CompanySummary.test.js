import { render } from "@testing-library/react";
import CompanySummary from "../CompanySummary";

test("render CompanySummary component as expected", () => {
  const props = {
    companyName: "Apple Inc",
    ticker: "AAPL",
    industry: "Technology",
    currentPrice: 120.62,
    openPrice: 119.9,
    prevClose: 120.53,
    high: 121.43,
    low: 119.675,
    marketCap: 2023469,
    ipo: "1980-12-12",
  };

  const { container } = render(<CompanySummary {...props} />);

  expect(container).toMatchSnapshot();
});
