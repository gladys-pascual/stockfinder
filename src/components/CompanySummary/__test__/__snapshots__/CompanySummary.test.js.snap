// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render CompanySummary component as expected 1`] = `
<div>
  <section
    class="company-summary"
  >
    <div
      class="summary-wrapper"
    >
      <div
        class="lhs"
      >
        <div
          class="company-header"
        >
          <h1
            class="ticker"
          >
            AAPL
          </h1>
          <div>
            <p
              class="company-name"
            >
              Apple Inc
            </p>
            <p
              class="industry"
            >
              Technology
            </p>
          </div>
        </div>
        <p
          class="current-price-label profit"
        >
          Current Price
        </p>
        <div
          class="price-and-pnl"
        >
          <h1
            class="current-price profit"
          >
            120.62
          </h1>
          <p
            class="profit pnl"
          >
            +0.6 %
          </p>
        </div>
      </div>
      <div
        class="rhs"
      >
        <div
          class="prices"
        >
          <div
            class="prices-lhs"
          >
            <p
              class="prices-heading"
            >
              OPEN
            </p>
            <p
              class="prices-text"
            >
              119.9
            </p>
          </div>
          <div
            class="prices-rhs"
          >
            <p
              class="prices-heading"
            >
              PREV CLOSE
            </p>
            <p
              class="prices-text"
            >
              120.53
            </p>
          </div>
        </div>
        <div
          class="prices"
        >
          <div
            class="prices-lhs"
          >
            <p
              class="prices-heading"
            >
              HIGH
            </p>
            <p
              class="prices-text"
            >
              121.43
            </p>
          </div>
          <div
            class="prices-rhs"
          >
            <p
              class="prices-heading"
            >
              LOW
            </p>
            <p
              class="prices-text"
            >
              119.675
            </p>
          </div>
        </div>
        <div
          class="prices"
        >
          <div
            class="prices-lhs"
          >
            <p
              class="prices-heading"
            >
              MARKET CAP
            </p>
            <p
              class="prices-text"
            >
              2.0 T
            </p>
          </div>
          <div
            class="prices-rhs"
          >
            <p
              class="prices-heading"
            >
              IPO
            </p>
            <p
              class="prices-text"
            >
              Dec 1980
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
